---
title: "Can Gene Expression Data Identify Patients With Inflammatory Bowel Disease?"
subtitle: "A Consulting Project for Math 6627 (3/3)"
author: "Benjamin Smith"
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
  pdf_document: 
    toc: true
---

# Introduction

(Quoted from SSC website.)

Inflammatory bowel disease (IBD), which is comprised of the two disease entities of Crohn’s disease (CD) and ulcerative colitis (UC), is an incurable gastrointestinal illness that results in chronic inflammation. IBD greatly affects patients’ quality of life. Approximately 1.5 million people have IBD in the United States and Canada, where the rates are among the highest in the world. 

There are currently no biomarkers for IBD, which could help to identify better treatments and individualize patient care. Such biomarkers could also be used to facilitate the development of clinical trials involving new medications. Recently, genome-wide association studies (GWAS) have significantly advanced our understanding about the importance of genetic susceptibility in IBD. Studies have identified a total of 201 IBD loci (Liu et al. 2015). However, these loci have yielded only a handful of candidate genes which often have small contributory effect in IBD. 

The aim of this case study is to construct classifiers for IBD using global gene expression data based on these candidate genes. The research questions are:

1. Can data features (i.e., variables or probesets or genes) be used to cluster individuals into three biological groups (i.e., healthy individuals, CD patients, UC patients)? 

2. Can data features (i.e., variables, probesets or genes) predict the disease state of individuals from three biological groups (i.e., healthy individuals, CD patients, UC patients)? 


# The Data

The data available consists of two datasets: 

(Quoted from SSC website.)

1. Global gene expression data: Burczynski et al. (2006) generated genome-wide gene expression profiles for 41 healthy individuals (note that the processed data includes only 41 individuals although the original study included 42 individuals), 59 CD patients, and 26 UC patients using Affymetrix HG-U133A human GeneChip array.  The GeneChip include approximate 22,000 probesets (each gene may have multiple probesets). The expression level of each probeset in each individual was quantified using MAS 5.0 software (we downloaded the processed data from ArrayExpress: E-GEOD-3365). 

2. IBD candidate genes: IBD candidate genes implicated in the 201 IBD associated loci were evaluated using GRAIL (Gene Relationships across Implicated Loci) and DAPPLE (Disease Association Protein-Protein Link Evaluator) software tools. A total of 225 unique genes (see Supplementary Table 9 of Liu et al. 2015) were identified, and 185 of these 225 genes are on the Affymetrix HG-U133A human GeneChip array.  These 185 candidate genes include 309 probesets.

Due to lack of domain knowledge and the ambiguity of the dataset, only the first data set will be used for this analysis. After the global gene expression data shaped properly for the analysis, there is no missing data present. With this taken care of, lets proceed with answering the two research questions. 

```{r message=FALSE, warning=FALSE, include=FALSE}

library(tidyverse)
library(reshape2)
library(readxl)
library(nlme)
library(glmnet)

dt1 <- read_excel("ConsultingData_3.xlsx", sheet=1) 
dt2 <- read_excel("ConsultingData_3.xlsx", sheet=2)

dtt2<-model.matrix(~Gene.Symbol, dt2)
# For now not using joined data
#joined_data<- dt1 %>% left_join(dt2)

dtt <- dt1 %>%
        t() %>% 
        as_tibble() %>% 
        setNames(dt1 %>% 
  t() %>% 
  as_tibble() %>% 
  slice(1:2) %>%
  as.list() %>% 
  lapply(function(x) x[!is.na(x)]) %>% 
  unname() %>%
  unlist()) %>% 
  slice(c(3:n()))

dtt<- dtt %>% 
      mutate(across(names(dtt)[-c(1,3,4)], ~as.numeric(.x)),
             Group = ifelse(Group ==  "Ulcerative" | Group =="Ulcerative Colitis", "Ulcerative Colitis",Group),
             Ethnicity =  ifelse( Ethnicity ==  "cacuasian" | Group =="caucasian", "caucasian",Ethnicity) )

#write.csv(dtt, "ConsultingData_3.csv") 
```



# Analysis


## Clustering Individuals Into Three Biological Groups

As is the case with the assignments in this practicum, the ability to thoroughly analyze the data under time constraint presents its challenges. On the SSC website it mentions that Liu, van Sommeren, Huang, et al found through association analyses which identify 38 susceptibility loci for inflammatory bowel disease and highlight shared genetic risk across populations. To explain that in terms for that people with a non-genetics background - the result found by Liu, van Sommeren, Huang, et al identifies 38 specific locations on an subjects chromosomes which highlight shared genetic risk across populations. 

An approach to cluster the data into three groups can be done through principal component analysis. Interestingly enough, ~ 80% of the variance in the data can be explained by 38 principal components as shown in the R output below: 

```{r message=FALSE, warning=FALSE, include=FALSE}
library(factoextra)
pca<- prcomp(data.matrix(dtt[,-1]), center = TRUE, scale = TRUE)

# We can use the first 38 principal components
pca_summary<-summary(pca)

# This agrees with the paper results
pca_summary$importance[,1:38]
```



A way to view the effectiveness of such a model would be to use 10-fold cross validation and see the mean-square error of prediction produced.  From figures 1 and 2 the MSE produced from PCA regression using 38 components produces a MSE of around 0.43 and the spread between predicted and measured groups is quite large. From this it can be seen that the 38 principal components are useful for explaining the variance in the data, however as a method for classifying individuals it is quite poor.

```{r echo=FALSE, message=FALSE, warning=FALSE,fig.cap="Mean Square Error of Prediction With PCA Regresion (38 Components)"}
<<<<<<< HEAD
library(pls)
library(caret)
set.seed(1)

pcr_model <- pcr(Group~ ., 
                 data = data.matrix(dtt) %>% as.data.frame.matrix(), 
                 scale = TRUE, 
                 validation = "CV",
                 ncomp=38)


validationplot(pcr_model, val.type=c("MSEP"),main="Mean Square Error of Prediction With PCA Regresion (38 Components)")

```


```{r echo=FALSE, message=FALSE, warning=FALSE,fig.cap="Predicted vs Measured Values using PCA Regression with 38 Components"}
predplot(pcr_model, main="Predicted vs Measured Values using PCA Regression with\n 38 Components")

```

While this method did prove to be a successful method for clustering data, it does demonstrate (and confirm) that the finding by  Liu, van Sommeren, Huang, et al identifies loci which are susceptible to receiving the diseases in question, but are not useful as predictors in of themselves. 

\newpage

## What Data Features Can Predict The Disease State From Three Biological Groups

With a focus on prediction, the use of multinomial regression with 10-fold cross validation is employed with a traditional machine learning approach where the data is split into training and testing sets with a 75%/25% training-testing split in the data. 

[ADD FIGURE CAPTIONS AND DESCRIPTIONS]

```{r}
set.seed(1)
dtt2 <- data.matrix(dtt)
y <- dtt2[,1]
X <- dtt2[,-1]

train = sample(seq(length(y)),75,replace=FALSE)
# 10 -fold crossvalidation
cv_model <- cv.glmnet(X, y, family='multinomial', alpha = 1)
best_lambda<- cv_model$lambda.min
```


```{r}
plot(cv_model,main=expression("Multinomial Deviance vs "*lambda[Group]*""))
```

```{r}
train_model <- glmnet(X[train,], y[train],family='multinomial', alpha = 1, lambda = best_lambda)

confusion.glmnet(train_model, newx = X[-train,], newy = y[-train])
```

```{r}
sparseEstimates<-data.frame(
       `Crohn's Disease`=train_model$beta$`1`[,],
       `Normal` =train_model$beta$`2`[,],
       `Ulcerative Colitis`=train_model$beta$`3`[,])

colnames(sparseEstimates)<-c("Crohn's Disease","Normal","Ulcerative Colitis")

sparseEstimates %>% 
  filter(!(`Crohn's Disease`==0 &`Normal`==0 & `Ulcerative Colitis`==0)) %>% 
  knitr::kable(caption = "Sparse Estimates for Multinomial Classification of Groups")
```





# Conclusion



# References

1. Statistical Society of Canada, Can Gene Expression Data Identify Patients With Inflammatory Bowel Disease? (2017). https://ssc.ca/en/meeting/annual/2017/case-study-2

2. Liu JZ, van Sommeren S, Huang H, et al. Association analyses identify 38 susceptibility loci for inflammatory bowel disease and highlight shared genetic risk across populations. Nat Genet. 47(9):979-86 (2015).

3. Ron LP, Natalie CT, Krystyna AZ, et al. Molecular classification of Crohn's disease and ulcerative colitis patients using transcriptional profiles in peripheral blood mononuclear cells. Michael E Burczynski, J Mol Diagn 8(1):51-61 (2006).

4. Ambroise C, McLachlan GJ. Selection bias in gene extraction on the basis of microarray gene-expression data. Proc Natl Acad Sci USA. 99(10):6562-6 (2002).

5. Dupuy A, Simon RM. Critical review of published microarray studies for cancer outcome and guidelines on statistical analysis and reporting. J Natl Cancer Inst. 99(2):147-57 (2007).

5. Cross-Validated, How to use R prcomp results for prediction? https://stats.stackexchange.com/questions/72839/how-to-use-r-prcomp-results-for-prediction

# Code Appendix

```{r}
=======
>>>>>>> 7e052d17e71b73e7e6975bc7e5a5cb7437654e0b
library(pls)
library(caret)
set.seed(1)

pcr_model <- pcr(Group~ ., 
                 data = data.matrix(dtt) %>% as.data.frame.matrix(), 
                 scale = TRUE, 
                 validation = "CV",
                 ncomp=38)


validationplot(pcr_model, val.type=c("MSEP"),main="Mean Square Error of Prediction With PCA Regresion (38 Components)")

```


```{r echo=FALSE, message=FALSE, warning=FALSE,fig.cap="Predicted vs Measured Values using PCA Regression with 38 Components"}
predplot(pcr_model, main="Predicted vs Measured Values using PCA Regression with\n 38 Components")

```

While this method did prove to be a successful method for clustering data, it does demonstrate (and confirm) that the finding by  Liu, van Sommeren, Huang, et al identifies loci which are susceptible to receiving the diseases in question, but are not useful as predictors in of themselves. 

\newpage

## What Data Features Can Predict The Disease State From Three Biological Groups

With a focus on prediction, the use of multinomial regression is employed
```{r}
dtt2 <- data.matrix(dtt)
y <- dtt2[,1]
X <- dtt2[,-1]

# 10 -fold crossvalidation
cv_model <- cv.glmnet(X, y, family='multinomial', alpha = 1)
best_lambda<- cv_model$lambda.min
```


```{r}
plot(cv_model,main="This needs a caption")
```

```{r}
best_model <- glmnet(X, y, alpha = 1, lambda = best_lambda)


sparseEstimates<- as.matrix(coef(best_model),rownames) %>% 
                  as.data.frame.matrix() %>%  
                  filter(s0!=0)

sparseEstimates[order(-abs(sparseEstimates$s0)),,drop=FALSE] %>%  knitr::kable(caption="Sparse Estimates")
```


```{r}
library(mclogit)
library(nnet)
library(memisc)

mod <- multinom(Group ~ .,data=dtt)
```



# Conclusion



# References

1. Statistical Society of Canada, Can Gene Expression Data Identify Patients With Inflammatory Bowel Disease? (2017). https://ssc.ca/en/meeting/annual/2017/case-study-2

2. Liu JZ, van Sommeren S, Huang H, et al. Association analyses identify 38 susceptibility loci for inflammatory bowel disease and highlight shared genetic risk across populations. Nat Genet. 47(9):979-86 (2015).

3. Ron LP, Natalie CT, Krystyna AZ, et al. Molecular classification of Crohn's disease and ulcerative colitis patients using transcriptional profiles in peripheral blood mononuclear cells. Michael E Burczynski, J Mol Diagn 8(1):51-61 (2006).

4. Ambroise C, McLachlan GJ. Selection bias in gene extraction on the basis of microarray gene-expression data. Proc Natl Acad Sci USA. 99(10):6562-6 (2002).

5. Dupuy A, Simon RM. Critical review of published microarray studies for cancer outcome and guidelines on statistical analysis and reporting. J Natl Cancer Inst. 99(2):147-57 (2007).

5. Cross-Validated, How to use R prcomp results for prediction? https://stats.stackexchange.com/questions/72839/how-to-use-r-prcomp-results-for-prediction

# Code Appendix